widget/gridwrap.go

```diff
@@ -317,6 +326,17 @@ func (l *GridWrap) UnselectAll() {
 	}
 }
 
+func (l *GridWrap) contentMinSize() fyne.Size {
+	padding := theme.Padding()
+	if lenF := l.Length; lenF != nil {
+		cols := l.ColumnCount()
+		rows := float32(math.Ceil(float64(lenF()) / float64(cols)))
+		return fyne.NewSize(l.itemMin.Width,
+			(l.itemMin.Height+padding)*rows-padding)
+	}
+	return fyne.NewSize(0, 0)
```

Comment:
I realise that this just was copied from the other method more or less but lets get the logic ordered up to our usual standard with edge-cases indented and main path inline:
```suggestion
	if l.Length == nil {
		return fyne.NewSize(0, 0)
	}
	
	cols := l.ColumnCount()
	rows := float32(math.Ceil(float64(l.Length()) / float64(cols)))
	return fyne.NewSize(l.itemMin.Width, (l.itemMin.Height+padding)*rows-padding)
```

Reply:
Interestingly, the old way was more thread-safe than the new way (user could unset the Length callback in between checking it for nil), but given that we are moving to the single-threaded model it's not an issue.

Reply:
Without actually using any locks they are both half broken but yeah the old one was better in that regard. You can always save the callback to a variable if you like.

Reply:
Nah, it won't be an issue at all with the new threading model :) (b/c you'll be disallowed from changing it while the logic here is running)

Reply:
But you have labelled this as going into v2.4.5?

Reply:
As you mentioned it was half-broken anyway, and unsetting the Length callback after having set it before for List is basically a developer error anyway (I can't think of a legitimate use case). I wouldn't worry about it, but if you want I can make the change